### PR TITLE
feat: my suggestions sort

### PR DESCRIPTION
**Pull request by trullse**

### CODE REVIEW

File: lib/src/presentation/utils/typedefs.dart [hunk 1]
```diff
@@ -21,4 +21,4 @@ typedef OnSaveToGalleryCallback = Future<bool?> Function(String url);
 typedef OnGetUserById = Future<SuggestionAuthor?> Function(String id);
 
 /// Sorting type values
-enum SortType { upvotes, creationDate }
+enum SortType { upvotes, creationDate, mySuggestions }
```
Comment (luffan) [inline lib/src/presentation/utils/typedefs.dart:24]: Change it to userSuggestion

File: lib/src/presentation/pages/suggestions/suggestions_state.dart [hunk 2]
```diff
@@ -136,6 +137,17 @@ extension SortTypeExtension on SortType {
       SortType.upvotes => (a, b) => b.upvotesCount.compareTo(a.upvotesCount),
       SortType.creationDate => (a, b) =>
           b.creationTime.compareTo(a.creationTime),
+      SortType.mySuggestions => (a, b) {
+          if (a.authorId == i.userId && b.authorId == i.userId) {
```
Comment (luffan) [inline lib/src/presentation/pages/suggestions/suggestions_state.dart:141]: remove this if

Comment (trullse) [reply lib/src/presentation/pages/suggestions/suggestions_state.dart]: It won't work the way it should. In this case user's suggestions won't sort by the votes count. I rewrote this code to make max number of conditions check less

File: lib/src/presentation/pages/suggestions/suggestions_state.dart [hunk 3]
```diff
@@ -136,6 +137,17 @@ extension SortTypeExtension on SortType {
       SortType.upvotes => (a, b) => b.upvotesCount.compareTo(a.upvotesCount),
       SortType.creationDate => (a, b) =>
           b.creationTime.compareTo(a.creationTime),
+      SortType.userSuggestion => (a, b) {
+          if (a.authorId == i.userId) {
```
[the remark this answers is not in the document]
Comment (trullse) [reply lib/src/presentation/pages/suggestions/suggestions_state.dart:141]: Yes, the same, rewrote in a new commit

**Pull request merged**